{
  log.info("Expected exception: " + e);
  if (!(e.getCause() instanceof IgniteClientDisconnectedException))   log.error("Unexpected cause: " + e.getCause(),e);
  assertTrue("Unexpected cause: " + e.getCause(),e.getCause() instanceof IgniteClientDisconnectedException);
  IgniteClientDisconnectedException e0=(IgniteClientDisconnectedException)e.getCause();
  assertNotNull(e0.reconnectFuture());
  return e0.reconnectFuture();
}
